### DISCUSSION TITLE
How to do a paired test with add_test_asterisks()

### DISCUSSION THREAD

**Discussion by hidekimiwa** [jbengler/tidyplots#32] [Ask a question] [answered]
I am trying to display p-values on a graph using add_test_asterisks(). However, the p-values displayed seem to differ from those calculated using Rcmdr or jamovi. Could this be due to differences in the calculation methods? Or is it only applicable to paired comparisons, such as the paired t-test or Wilcoxon's signed rank test? In other words, is it not possible to use this function for independent comparisons, such as the independent t-test or Mann-Whitney U test?

**Comment by jbengler** [comment]
The `add_test_asterisks()` and `add_test_pvalue()`functions are wrappers around the ggpubr function `ggpubr::geom_pwc()`. You can find the documentation here: https://rpkgs.datanovia.com/ggpubr/reference/geom_pwc.html

If you have a concrete example to investigate, feel free to provide a reproducible example (reprex) containing the data and the code here.

**Comment by hidekimiwa** [comment]
Thank you for your response.
I tried based on the information you provided, but it seems that tests for paired samples cannot be performed by default. Even when using "wilcox_test" or "t_test" with paired = TRUE, the p-values were the same as when paired = FALSE. Additionally, the p-values differed from those obtained with pairwise_wilcox_test.
I think I’ll use ggplot2 or similar to display p-values or asterisks on the graphs instead of using tidyplots.

**Comment by jbengler** [comment]
Which package for ggplot2 are you using to display p-values or asterisks? Just for me to know, in case I want to switch from ggpubr to something else in the future.

**Comment by hidekimiwa** [comment]
I then tried to display the p-values obtained from rstatix::pairwise_wilcox_test using ggpubr::stat_pvalue_manual, but unfortunately, it is not working as expected at the moment. I would greatly appreciate any advice if possible.

**Comment by jbengler** [comment, marked as answer]
The combination of `paired = TRUE` and `ggpubr::stat_pvalue_manual()` is a little bit tricky.

This works:

```
library(tidyverse)
library(tidyplots)

data <- 
  study %>% 
  dplyr::filter(dose == "high")

stats <- 
  data %>% 
  dplyr::arrange(participant) %>% 
  rstatix::wilcox_test(score ~ group, paired = TRUE) %>% 
  rstatix::add_significance() %>% 
  rstatix::add_xy_position()

# tidyplot solution
data %>% 
  tidyplot(x = group, y = score, color = group, fill = NA) %>% 
  add_line(group = participant, color = "lightgrey") %>% 
  add_data_points() %>% 
  add(ggpubr::stat_pvalue_manual(stats, size = 7/.pt))

# ggplot solution
data %>% 
  ggplot(aes(x = group, y = score, color = group)) +
  geom_line(aes(group = participant), color = "lightgrey") +
  geom_point() +
  ggpubr::stat_pvalue_manual(stats)
```

**Caution:** The pairing of the samples depends of their **order** in the data frame. I always found this a very dangerous thing to rely on.

And if you prefer asterisks over p-values:

```
# tidyplot solution
data %>% 
  tidyplot(x = group, y = score, color = group, fill = NA) %>% 
  add_line(group = participant, color = "lightgrey") %>% 
  add_data_points() %>% 
  add(ggpubr::stat_pvalue_manual(stats, label = "p.signif", size = 7/.pt))

# ggplot solution
data %>% 
  ggplot(aes(x = group, y = score, color = group)) +
  geom_line(aes(group = participant), label = "p.signif", color = "lightgrey") +
  geom_point() +
  ggpubr::stat_pvalue_manual(stats)
```

If this solves your problem, could you hit the "Mark as answer" button below? Thanks!

Best
Jan

**Reply by hidekimiwa** [reply]
It works! Thank you so much!

**Reply by jbengler** [reply]
Hi @hidekimiwa
That is great! Could you mark my post above as answer using the "Mark as answer" button.
That makes it easier for others to quickly find a solution when searching the Discussions.
Thanks!
